### ISSUE TITLE
Document Unit Tests do not actually work

### ISSUE THREAD

**Issue by Peilin-Yang** [castorini/Anserini#383]
Currently, the unit tests are something like 
```
for (int i = 0; i < rawFiles.size(); i++) {
      BaseFileSegment<TrecCollection.Document> iter = collection.createFileSegment(rawFiles.get(i));
      while (true) {
        try {
          TrecCollection.Document parsed = iter.next();
          assertEquals(parsed.id(), expected.get(i).get("id"));
          assertEquals(parsed.content(), expected.get(i).get("content"));
        } catch (NoSuchElementException e) {
          break;
        }
      }
    }
```
The execution path directly goes to the `catch` branch w/o actually test the parsing document branch. 
@borislin Is this part of your previous refactor?

**Comment by borislin** [comment]
@Peilin-Yang 

Yes, this is a bug caused by my previous refactor.
I've fixed this bug in #384. Please review.

**Comment by Peilin-Yang** [comment]
Merged #384
